### PR TITLE
UF-10671 - Add mentor to employee checklists

### PR DESCRIPTION
**Pull request by jklingst**
Adds a mentor object, holding user id and name, to employee checklists.

## Types of changes

- [ ] Bug fix
- [x] New feature
- [ ] Removed feature
- [ ] Code style update (formatting etc.)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes

## Does this PR introduce a breaking change?
- [ ] Yes (I have stepped the version number accordingly)
- [x] No
      
## Checklist:

- [x] My code follows the code style of this project.
- [x] I have updated the documentation accordingly (if applicable).
- [x] I have added/updated tests to cover my changes (if applicable).

### CODE REVIEW

File: src/main/java/se/sundsvall/checklist/api/EmployeeChecklistResource.java [hunk 1]
```diff
@@ -224,4 +227,42 @@ ResponseEntity<EmployeeChecklistResponse> initiateSpecificEmployeeChecklist(
...
 		return ok().body(employeeChecklistService.initiateSpecificEmployeeChecklist(municipalityId, personId));
 	}
+
+	@Operation(
+		summary = "Set the mentor",
+		description = "Set the mentor on a specific employee checklist",
+		responses = {
+			@ApiResponse(responseCode = "200", description = "Successful Operation", useReturnTypeSchema = true)
+		})
+	@PostMapping(
```
Comment (josc04) [inline src/main/java/se/sundsvall/checklist/api/EmployeeChecklistResource.java:237]: PutMapping (as it updates/replaces if mentor already exists and adds it if it doesn exist)?

**Review by AndreSoderlund — DISMISSED** [review]

File: src/main/java/se/sundsvall/checklist/api/model/Mentor.java [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+import lombok.Builder;
+import lombok.Data;
+import lombok.NoArgsConstructor;
+
+@Data
+@NoArgsConstructor
+@AllArgsConstructor(access = AccessLevel.PACKAGE)
+@Builder(setterPrefix = "with")
+@Schema(description = "Model for a mentor on an employee checklist")
+public class Mentor {
```
Comment (josc04) [inline src/main/java/se/sundsvall/checklist/api/model/Mentor.java:15]: For each attribute,:

-  add `@Schema(description = "<description>", example = "<example>")`
- validate not empty?

File: src/test/java/se/sundsvall/checklist/integration/db/EmployeeChecklistIntegrationTest.java [hunk 3]
```diff
@@ -675,6 +677,35 @@ void deleteEmployeeChecklist() {
...
+		var employeeChecklistId = UUID.randomUUID().toString();
+		var entity = EmployeeChecklistEntity.builder().build();
+		var mentor = Mentor.builder().build();
+
+		when(employeeChecklistsRepositoryMock.findByIdAndChecklistMunicipalityId(employeeChecklistId, municipalityId)).thenReturn(Optional.of(entity));
+
+		integration.setMentor(municipalityId, employeeChecklistId, mentor);
+
+		verify(employeeChecklistsRepositoryMock).findByIdAndChecklistMunicipalityId(employeeChecklistId, municipalityId);
+		verify(employeeChecklistsRepositoryMock).save(entity);
```
Comment (josc04) [inline src/test/java/se/sundsvall/checklist/integration/db/EmployeeChecklistIntegrationTest.java:692]: Use captor and verify that mentor on captured entity is equal to mentor?

File: src/test/java/se/sundsvall/checklist/integration/db/EmployeeChecklistIntegrationTest.java [hunk 4]
```diff
@@ -675,6 +677,35 @@ void deleteEmployeeChecklist() {
...
+		var municipalityId = "municipalityId";
+		var employeeChecklistId = UUID.randomUUID().toString();
+		var entity = EmployeeChecklistEntity.builder().build();
+
+		when(employeeChecklistsRepositoryMock.findByIdAndChecklistMunicipalityId(employeeChecklistId, municipalityId)).thenReturn(Optional.of(entity));
+
+		integration.deleteMentor(municipalityId, employeeChecklistId);
+
+		verify(employeeChecklistsRepositoryMock).findByIdAndChecklistMunicipalityId(employeeChecklistId, municipalityId);
+		verify(employeeChecklistsRepositoryMock).save(entity);
```
Comment (josc04) [inline src/test/java/se/sundsvall/checklist/integration/db/EmployeeChecklistIntegrationTest.java:706]: Use captor to verify that no mentor is set on captured entity?

File: src/main/java/se/sundsvall/checklist/api/model/Mentor.java (hunk 2 above)
Comment (jklingst) [reply src/main/java/se/sundsvall/checklist/api/model/Mentor.java:15]: Definitely yes on the `@Schema` annotations - they should indeed have been added already.

Regarding not empty, I maybe misinterpreted the information when told that the fields are optional. But sure, it makes sense not allowing blank values to be passed.

Comment (josc04) [reply src/main/java/se/sundsvall/checklist/api/model/Mentor.java:15]: My interpretation is that mentor is optional, but if present then all values should be present. Verify how it should be with Jakob?

**Review by josc04 — APPROVED** [review]

**Pull request merged**